echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/PipelineTriggerAction.groovy

```diff
@@ -27,21 +29,33 @@ import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
 class PipelineTriggerAction extends ActionSupport {
-    static final Logger LOGGER = LoggerFactory.getLogger(PipelineTriggerAction)
-
-    @Override
-    void execute(Context context, Execution execution) throws Exception {
-        try {
-            def pipelineInitiator = (PipelineInitiator) ActionDependencies.getBean(PipelineInitiator)
-            def pipelineCache = (PipelineCache) ActionDependencies.getBean(PipelineCache)
-            def pipeline = PipelineTriggerConverter.fromParameters(pipelineCache, context.parameters)
-
-            LOGGER.info("Executing PipelineTriggerAction for '${pipeline}'...")
-            pipelineInitiator.call(pipeline)
-            LOGGER.info("Successfully executed PipelineTriggerAction for '${pipeline}")
-        } catch (Exception e) {
-            LOGGER.error("Exception occurred while executing PipelineTriggerAction", e)
-            throw e
-        }
+  static final Logger LOGGER = LoggerFactory.getLogger(PipelineTriggerAction)
+
+  private final Registry registry
+
+  PipelineTriggerAction(Registry registry) {
+    this.registry = registry
+  }
+
+  @Override
+  void execute(Context context, Execution execution) throws Exception {
+    try {
+      def pipelineInitiator = (PipelineInitiator) ActionDependencies.getBean(PipelineInitiator)
+      def pipelineCache = (PipelineCache) ActionDependencies.getBean(PipelineCache)
+      def pipeline = PipelineTriggerConverter.fromParameters(pipelineCache, context.parameters)
+
+      LOGGER.info("Executing PipelineTriggerAction for '${pipeline}'...")
+      pipelineInitiator.call(pipeline)
+
+      Id id = registry.createId("pipelines.triggered")
+        .withTag("monitor", getClass().getSimpleName())
+        .withTag("application", Optional.ofNullable(pipeline.getApplication()).orElse("NULL_APPLICATION"))
+      registry.counter(id).increment()
+
+      LOGGER.info("Successfully executed PipelineTriggerAction for '${pipeline}")
+    } catch (Exception e) {
+      LOGGER.error("Exception occurred while executing PipelineTriggerAction", e)
```

Comment:
while we're at it, let's have a metric for pipeline trigger errors if you don't mind

Reply:
:D I checked no errors have been seen in the last week. You're right, though, probably doesn't hurt to have a metric